config.yaml

```diff
@@ -46,7 +46,19 @@ options:
   addons-registry:
     type: string
     default: ""
-    description: Specify the docker registry to use when applying addons
+    description: |
+      Specify the docker registry to use when applying addons.
+
+      DEPRECATED in 1.15: Use the broader 'image-registry' config option instead. If both
+      options are set, addons-registry will take precedence during cdk-addons snap
+      configuration until v1.14 reaches EOL.
```

Comment:
Should we do this or make addons-registry takes precedent in < 1.15? Seems it would be an easy check and also not rely on us later remembering to change it.

Reply:
@hyperbolic2346 i kept `addons-registry` taking precedent in case people already had it set in their deployment.  If it's empty (the default), `image-registry` will take over -- if it's not empty, we'll only use it for addons, with the rest of the images (pause, default backend, etc) still expected to come from `image-registry`.

Reply:
No, I get that and like it. My question is in the "this takes precedent until we change it after 1.14 is EOL" vs "this takes precedent if the api server version is < 1.15". It is a subtle difference of hard-coded to always use addons-registry first or prefer different things based on api server version. Maybe it should be if api server version is < 1.17 to allow upgrade time? I certainly don't want to yank the rug out from under people.

Reply:
ahhh, roger that @hyperbolic2346.  I've updated the config description and added a 1.17 check to address this.